config/v1/types_cluster_version.go

```diff
@@ -42,11 +42,16 @@ type ClusterVersionSpec struct {
 
 	// desiredUpdate is an optional field that indicates the desired value of
 	// the cluster version. Setting this value will trigger an upgrade (if
-	// the current version does not match the desired version). The set of
-	// recommended update values is listed as part of available updates in
-	// status, and setting values outside that range may cause the upgrade
-	// to fail. You may specify the version field without setting image if
-	// an update exists with that version in the availableUpdates or history.
+	// the current architecture or version does not match the desired
+	// architecture or version). The set of recommended update values is
+	// listed as part of available updates in status, and setting values
+	// outside that range may cause the upgrade to fail. You may specify the
+	// version field without setting architecture or image if an update exists
+	// with that version in the availableUpdates or history. Upgrading from a
+	// homogeneous to a heterogeneous cluster architecture at the same version
+	// can be done by setting architecture to "multi". Image can be used to
+	// upgrade from a homogeneous to a heterogeneous cluster architecture at a
```

Comment:
I'd rather drop this sentence, because it's a "this is possible" safety valve, and not a "this is encouraged" UX.